pkg/sync/sync.go

```diff
@@ -96,23 +89,26 @@ func makeTar(srcPath, destPath string, writer io.Writer, files []string, globExp
 
 				// We use "FromSlash" to make this OS-based (Windows uses \, Linux & macOS use /)
 				// we get the relative path by joining the two
-				srcFile, err := filepath.Rel(filepath.FromSlash(srcPath), filepath.FromSlash(fileAbsolutePath))
+				destFile, err := filepath.Rel(filepath.FromSlash(srcPath), filepath.FromSlash(fileAbsolutePath))
 				if err != nil {
 					return err
 				}
 
 				// Now we get the source file and join it to the base directory.
-				srcFile = filepath.Join(filepath.Base(srcPath), srcFile)
+				srcFile := filepath.Join(filepath.Base(srcPath), destFile)
+
+				glog.V(4).Infof("makeTar srcFile: %s", srcFile)
+				glog.V(4).Infof("makeTar destFile: %s", destFile)
 
 				// The file could be a regular file or even a folder, so use recursiveTar which handles symlinks, regular files and folders
-				err = recursiveTar(filepath.Dir(srcPath), srcFile, filepath.Dir(destPath), srcFile, tarWriter, globExps)
+				err = recursiveTar(filepath.Dir(srcPath), srcFile, filepath.Dir(destPath), destFile, tarWriter, globExps)
 				if err != nil {
 					return err
 				}
 			}
 		}
 	} else {
-		return recursiveTar(filepath.Dir(srcPath), filepath.Base(srcPath), filepath.Dir(destPath), filepath.Base(destPath), tarWriter, globExps)
+		return recursiveTar(filepath.Dir(srcPath), filepath.Base(srcPath), filepath.Dir(destPath), "", tarWriter, globExps)
```

Comment:
So odo 1 currently tar'd local projects files on your laptop/workspace to `/projects` in the container, but it did in a way that the archvie had a top level dir. For example, if your local laptop current dir was `someproject` where you ran odo push, then tar would create `someproject/projectfiles` and did an tar extract with command `cmdArr := []string{"tar", "xf", "-", "-C", targetPath, "--strip", "1"}` where strip was basically stripping this top level dir created during tar, so that inside the container, files would be in `/projects`

This was fine with kube exec api, but docker client api `CopyToContainer` does not handle this. The files are copied to `/projects/someproject/<files>` and would need to execute another command to handle this.

So instead of creating the destination tar in `filepath.Base(destPath)`, we just do not specify a dir, and pass in `""` for this reason